nats-base-client/util.ts

```diff
@@ -200,3 +202,37 @@ export function shuffle(a: any[]): any[] {
   }
   return a;
 }
+
+export function encodeHeader(headers: Headers): Uint8Array {
+  let v = `NATS/1.0\r\n`;
+  for (const [key, value] of headers) {
+    v += `${key}: ${value}\r\n`;
+  }
+  v += `\r\n`;
+  return new TextEncoder().encode(v);
+}
+
+export function decodeHeaders(buf: Uint8Array): Headers {
+  const headers = new Headers();
+  const v = new TextDecoder("utf-8").decode(buf);
+  let raw = v.split("\r\n");
+  let proto = raw[0];
+  const code = proto.replace("NATS/1.0 ", "");
+  if (code !== "") {
+    // this is only currently a timeout
```

Comment:
I don't understand what you are doing here.

Reply:
parsing the error - removed the NATS/1.0, and if it is not empty, parsing a number out.